timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -0,0 +1,319 @@
+        prediction_length: int = 1,
+        path: Optional[str] = None,
+        name: Optional[str] = None,
+        eval_metric: str = None,
+        hyperparameters: Dict[str, Any] = None,
+        **kwargs,  # noqa
+    ):
+        hyperparameters = hyperparameters if hyperparameters is not None else {}
+
+        model_path_input = hyperparameters.get("model_path", self.default_model_path)
+        self.model_path = MODEL_ALIASES.get(model_path_input, model_path_input)
+
+        # TODO: automatically determine batch size based on GPU / memory availability
+        self.batch_size = hyperparameters.get("batch_size", self.default_batch_size)
+        self.num_samples = hyperparameters.get("num_samples", self.default_num_samples)
+        self.device = hyperparameters.get("device")
+        self.torch_dtype = hyperparameters.get("torch_dtype", "auto")
+        self.data_loader_num_workers = hyperparameters.get("data_loader_num_workers", 1)
+        self.optimization_strategy: Optional[Literal["onnx", "ovm"]] = hyperparameters.get(
+            "optimization_strategy", None
+        )
+        self.context_length = hyperparameters.get("context_length")
+
+        if self.context_length is not None and self.context_length > self.maximum_context_length:
+            logger.warning(
+                f"Context length {self.context_length} exceeds maximum context length {self.maximum_context_length}."
+                f"Context length will be set to {self.maximum_context_length}."
+            )
+            self.context_length = self.maximum_context_length
+
+        model_path_safe = str.replace(model_path_input, "/", "__")
+        name = (name if name is not None else "Chronos") + f"[{model_path_safe}]"
+
+        super().__init__(
+            path=path,
+            freq=freq,
+            prediction_length=prediction_length,
+            name=name,
+            eval_metric=eval_metric,
+            hyperparameters=hyperparameters,
+            **kwargs,
+        )
+
+        self.model_pipeline: Optional[Any] = None  # of type OptimizedChronosPipeline
+
+    def save(self, path: str = None, verbose: bool = True) -> str:
+        pipeline = self.model_pipeline
+        self.model_pipeline = None
+        path = super().save(path=path, verbose=verbose)
+        self.model_pipeline = pipeline
+
+        return str(path)
+
+    @classmethod
+    def load(cls, path: str, reset_paths: bool = True, verbose: bool = True) -> "ChronosModel":
+        model = load_pkl.load(path=os.path.join(path, cls.model_file_name), verbose=verbose)
+        if reset_paths:
+            model.set_contexts(path)
+        return model
+
+    def score_and_cache_oof(
+        self,
+        val_data: TimeSeriesDataFrame,
+        store_val_score: bool = False,
+        store_predict_time: bool = False,
+    ) -> None:
+        super().score_and_cache_oof(val_data, store_val_score, store_predict_time)
+
+    def _is_gpu_available(self) -> bool:
```

Comment:
Nit: Should we move this method to `AbstractTimeSeriesModel` since multiple models rely on this method (also `AbstractGluonTSModel`)?

Reply:
It's already there, returning `False` by default. I guess the idea was to not make mention of torch if it's not needed.

Reply:
I see. I guess that in the ideal world it would be fine to use the `torch` import inside this method since only models that support GPU will ever call`_is_gpu_available()`. However, because of the idiosyncrasies of the HPO logic with ray this method is actually called by all models. I'm fine keeping it as is, or maybe adding a TODO comment if you feel like it (to remove `_is_gpu_available` from HPO logic and use a tag indicating if model supports GPU instead).